array.go

```diff
@@ -967,3 +967,43 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// Transform runs the passed function on all the Elements in the array
+// returns a new array without effeecting original array.
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{"foo", "bar"})
+//	transformedArray := array.Transform(
+//		func(index int, value interface{}) interface{} {
+//			return transformedValue.(interface{})
+//		},
+//	)
+//	transformedArray.Equals([]interface{}{....})
```

Comment:
Could you please make this example a bit more complete? I mean, perform some real transformation in func (e.g., apply strings.ToUpper) and check for real values in transformedArray.Equals.

Reply:
Understood.